libs/wingc/src/type_check.rs

```diff
@@ -12,7 +13,70 @@ use type_env::TypeEnv;
 
 use self::type_env::StatementIdx;
 
+pub struct UnsafeRef<T>(*const T);
```

Comment:
If we keep this up I feel like we're going to reinvent RefCell 😁

Reply:
Of course this isn't a `RefCell`, which is safe. This is more of a curtain hiding the fact that I'm doing risky things under the hood. I wanted this because I thought the `IdentifierKind` might have the same issues as the Type (require cross reference) and I didn't want to duplicate the code. But eventually I didn't need it.
Then I just used the same logic to hide the unsafe `parent` pointer in the `TypeEnv`.

Reply:
Yeah, I totally understand the intent, was mostly a joke 👍